airflow/models/dag.py

```diff
@@ -1658,9 +1679,13 @@ def set_task_instance_state(
 
         task = self.get_task(task_id)
         task.dag = self
+        if not map_indexes:
+            map_indexes = [-1]
```

Comment:
Intuitively I think not passing `map_idnexes` should default to setting states of all task instances of this task, not just -1 (matters if the task is mapped).

Another thing to consider is that, since this function is named `set_task_instance_state` (singular), it probably should only be able to set state to one task instance, i.e. it should only take one map index, not multiple. If we want to allow setting multiple mapped task instances, we should name the function `set_task_state` and deprecate this name.